dawn/src/dawn/CodeGen/CXXNaive-ico/ASTStencilBody.cpp

```diff
@@ -376,15 +376,22 @@ void ASTStencilBody::visit(const std::shared_ptr<ast::ReductionOverNeighborExpr>
                "lhs, auto "
         << ASTStencilBody::ReductionIndexVarName(reductionDepth_ + 1)
         << ", auto const& weight) mutable {\n";
-    ss_ << "lhs " << expr->getOp() << "= ";
-    ss_ << "weight * ";
   } else {
     ss_ << ", [&, " + ASTStencilBody::ReductionSparseIndexVarName(reductionDepth_) +
                " = int(0)](auto& lhs, auto "
         << ASTStencilBody::ReductionIndexVarName(reductionDepth_ + 1) << ") mutable { ";
+  }
+
+  if(expr->getOp() != "+") {
```

Comment:
I don't think this is safe. There are other compound assignments (e.g. `-=`)

Reply:
I thought at the moment there is only `sum_over`, `max_over` and `min_over`?
In that context `ReductionOverNeighborExpr` it should only be these 3.

I guess I could try to make it more future proof?
Would it be good enough if I excluse `+`, `-`, `*`, `/` and `%`?

Reply:
`sum_over`, `max_over` and `min_over` are the shortcuts available in dusk. there is also a more general `reduce_over` which takes an Operator as third argument. I'm currently not sure we ever formally specified which operators are allowed for the reduction concept. Your selection seems fine to me